prompt/prompt-executor/prompt-executor-clients/src/commonMain/kotlin/ai/koog/prompt/executor/clients/InternalAPI.kt

```diff
@@ -0,0 +1,17 @@
+package ai.koog.prompt.executor.clients
+
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This API is internal in koog and should not be used. It could be removed or changed without notice."
+)
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.TYPEALIAS,
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.PROPERTY,
+    AnnotationTarget.FIELD,
+    AnnotationTarget.CONSTRUCTOR,
+    AnnotationTarget.PROPERTY_SETTER,
+    AnnotationTarget.PROPERTY_SETTER
+)
+public annotation class InternalAPI
```

Comment:
Let's rename it to `InternalLLMClientApi` to better highlight the scope and purpose. We already have `InternalAgentToolsApi` annotation, so let's continue with this scoped annotations approach for now. We can introduce some single global `InternalKoogAPI` annotation later.

Note for maintainers: we will need to update custom inspection `MissingKDocForPublicAPI.inspection.kts` so that it doesn't require KDocs on public symbols marked with this annotation.